server/jetstream_cluster_2_test.go

```diff
@@ -6391,41 +6395,90 @@ func TestJetStreamClusterRePublishUpdateNotSupported(t *testing.T) {
 			if resp.Type != JSApiStreamUpdateResponseType {
 				t.Fatalf("Invalid response type %s expected %s", resp.Type, JSApiStreamUpdateResponseType)
 			}
-			if !IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
-				t.Fatalf("Expected error regarding config error, got %+v", resp.Error)
+			if IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
+				t.Fatalf("Expected no error regarding config error, got %+v", resp.Error)
 			}
 		}
-		expectFailUpdate()
 
-		// Now try with a new stream with RePublish present and then try to change config
-		cfg = &StreamConfig{
-			Name:     stream + "_2",
-			Storage:  MemoryStorage,
-			Replicas: replicas,
-			RePublish: &RePublish{
-				Source:      ">",
-				Destination: "bar.>",
-			},
+		expectRepublished := func(expectedRepub bool) {
+			t.Helper()
+
+			nc, js := jsClientConnect(t, s)
+			defer nc.Close()
+
+			var foos, bars atomic.Int32
+
+			// Create a subscriber for foo.> so that we can see
+			// our published message being echoed back to us.
+			_, err := js.Subscribe(
+				"foo.>",
+				func(msg *nats.Msg) { foos.Add(1) },
+				nats.AckNone(),
+				nats.DeliverNew(),
+			)
+			require_NoError(t, err)
+
+			// Create a subscriber for bar.> so that we can see
+			// any potentially republished messages.
+			_, err = js.Subscribe(
+				"bar.>",
+				func(msg *nats.Msg) { bars.Add(1) },
+				nats.AckNone(),
+				nats.DeliverNew(),
+			)
+			require_NoError(t, err)
+
+			// Publish a message, it will hit the foo.> stream and
+			// may potentially be republished to the bar.> stream.
+			_, err = js.Publish("foo."+stream, []byte("HELLO!"))
+			require_NoError(t, err)
+
+			// Wait for a little while so that we have enough time
+			// to determine whether it's going to arrive on one or
+			// both streams.
+			time.Sleep(time.Millisecond * 100)
+			switch {
+			case foos.Load() != 1:
```

Comment:
Same question here? We generally support 2 version back for server, so for now go 1.18 and go 1.19.